pkg/proxy/util/nodeport_addresses.go

```diff
@@ -51,17 +52,25 @@ func NewNodePortAddresses(family v1.IPFamily, cidrStrings []string) *NodePortAdd
 		}
 	}
 	if len(npa.cidrStrings) == 0 {
-		if family == v1.IPv4Protocol {
-			npa.cidrStrings = []string{IPv4ZeroCIDR}
+		if primaryIP == nil {
+			if family == v1.IPv4Protocol {
+				npa.cidrStrings = []string{IPv4ZeroCIDR}
+			} else {
+				npa.cidrStrings = []string{IPv6ZeroCIDR}
+			}
 		} else {
-			npa.cidrStrings = []string{IPv6ZeroCIDR}
+			if family == v1.IPv4Protocol {
+				// npa.cidrStrings = []string{(fmt.Sprintf("%s/%d", primaryIP.String(), len)}
+				npa.cidrStrings = []string{(&net.IPNet{IP: primaryIP, Mask: net.CIDRMask(32, 32)}).String()}
```

Comment:
I guess the simple thing here would just be to keep it as separate ipv4 and ipv6 statements:

```
	if family == v1.IPv4Protocol {
		npa.cidrStrings = []string{fmt.Sprintf("%s/32", primaryIP.String())}
	} else {
		npa.cidrStrings = []string{fmt.Sprintf("%s/128", primaryIP.String())}
	}
```

`net.IP` is slightly weird and it would be more annoying to extract the correct length from that